src/test/unit/lambda_tracer/test_lambda_reporter.py

```diff
@@ -28,42 +35,190 @@
 from lumigo_tracer.lumigo_utils import Configuration, InternalState
 
 
-@pytest.fixture
-def dummy_span():
-    return {"dummy": "dummy"}
-
-
-@pytest.fixture
-def function_end_span():
-    return {"dummy_end": "dummy_end"}
-
-
-@pytest.fixture
-def error_span():
-    return {"dummy": "dummy", "error": "Error"}
-
-
-def test_create_request_body_default(dummy_span):
+@fixture
+def dummy_span() -> dict:
+    return {"dummy": "dummy", "type": ENRICHMENT_TYPE}
+
+
+@fixture
+def function_end_span() -> dict:
+    return {"dummy_end": "dummy_end", "type": FUNCTION_TYPE, "envs": {"a": "b"}}
+
+
+@fixture
+def function_end_span_metadata() -> dict:
+    return {"dummy_end": "dummy_end", "type": FUNCTION_TYPE}
+
+
+@fixture
+def error_span() -> dict:
+    return {"dummy": "dummy", "error": "Error", "type": HTTP_TYPE}
+
+
+@fixture
+def redis_span() -> dict:
+    now = datetime.now()
+    return {
+        "type": REDIS_SPAN,
+        "id": "77d0b751-9496-4257-b910-25e33c029365",
+        "connectionOptions": {"host": "lumigo", "port": None},
+        "transactionId": "transaction-id",
+        "started": (now - timedelta(seconds=10)).timestamp() * 1000,
+        "requestArgs": '[[{"a": 1}], ["a"]]',
+        "lambda_container_id": "b7bfabd6-bc95-445a-965d-513922afbcdd",
+        "account": "account-id",
+        "region": "UNKNOWN",
+        "parentId": "parent-id",
+        "info": {"tracer": {"version": "1.1.230"}, "traceId": {"Root": ""}},
+        "requestCommand": '["SET", "GET"]',
+        "token": "t_token",
+        "ended": now.timestamp() * 1000,
+        "response": '"Result"',
+    }
+
+
+@fixture
+def redis_span_metadata(redis_span: dict) -> dict:
+    span_copy = copy.deepcopy(redis_span)
```

Comment:
But I want to check `the get_span_metadata` logic.

Reply:
do you want to test `get_span_metadata` logic or `_create_request_body`?
the only place that uses this logic executes `_create_request_body`, and not `get_span_metadata`.

If your goal is to test `get_span_metadata`, then call it directly. 
Otherwise, you only want to test that `get_span_metadata` is being called internally - i.e. you should use it to create the expected outcome.

Reply:
I'll clearify that I think that all this one-time-used fixtures should be deleted, and you should use `parameterized` instead of a single-lined-test that calls a util :)